### PR TITLE
[Pipelines] Migrate to 1ES pipeline

### PR DESCRIPTION
**Pull request by AnatolyPristensky**
Migrating Pipelines to 1ES Template. 

[AB#104414](https://dev.azure.com/msmobilecenter/Mobile-Center/_boards/board/t/Mobile%20SDK%20team/Backlog%20Items/?workitem=104414)

Successful build: https://dev.azure.com/msmobilecenter/SDK/_build/results?buildId=1521449&view=results

### CODE REVIEW

File: .azuredevops/appcenter-sdk-unity-build-release-1es.yml [hunk 1]
```diff
@@ -13,75 +26,83 @@ resources:
...
-            arguments: '-Target="RegisterUnity" -UnitySerialNumber="$(UNITY_SERIAL_NUMBER_WINDOWS)" -UnityUsername="$(UNITY_USERNAME)" -UnityPassword="$(UNITY_PASSWORD)" -Verbosity="Diagnostic"'
-
+            arguments: '-Target="RegisterUnity" -UnitySerialNumber="$(appcenter-sdk-unity-serial-number-windows)" -UnityUsername="$(appcenter-sdk-unity-username)" -UnityPassword="$(appcenter-sdk-unity-password)" -Verbosity="Diagnostic"'
         - script: |
-          echo After activating unity, some important .meta files are deleted, so they must be un-deleted
-          git config --global user.email "you@example.com"
-          git config --global user.name "Your Name"
-          git stash
+            echo After activating unity, some important .meta files are deleted, so they must be un-deleted
+            git config --global user.email "you@example.com"
```
Comment (DmitriyKirakosyan) [inline .azuredevops/appcenter-sdk-unity-build-release-1es.yml]: I don't think we need to setup user info for git, we are not commiting anything

File: .azuredevops/appcenter-sdk-unity-build-release-1es.yml [hunk 2]
```diff
@@ -13,75 +26,83 @@ resources:
...
+            git config --global user.name "Your Name"
+            git stash
           displayName: 'git stash'
-
+# Removing unnecessary file to prevent Binskim failure. This file used for tests and not required for build.
+        - script: |
+            git submodule init
+            git submodule update
+            rm BreakpadSupport/breakpad-support/src/main/cpp/google-breakpad/src/tools/windows/dump_syms/testdata/dump_syms_regtest64.exe
+          displayName: 'git submodule'
```
Comment (DmitriyKirakosyan) [inline .azuredevops/appcenter-sdk-unity-build-release-1es.yml]: Let's use more human readable display names, eg "Fetch submodules".
Removing the file can be done in a separate task with its own title.

**Review by DmitriyKirakosyan — DISMISSED** [review]

File: .azuredevops/appcenter-sdk-unity-build-release-1es.yml [hunk 3]
```diff
@@ -4,6 +4,19 @@ trigger:
 pr:
 - master
 - develop
+
+variables: 
+- name: EDITOR_URL_WIN
```
Comment (lucen-ms) [inline .azuredevops/appcenter-sdk-unity-build-release-1es.yml:9]: Where is this variable used?

File: .azuredevops/appcenter-sdk-unity-build-release-1es.yml [hunk 4]
```diff
@@ -4,6 +4,19 @@ trigger:
...
+  value: 'https://download.unity3d.com/download_unity/61fce66342ad/Windows64EditorInstaller/UnitySetup64-2018.4.18f1.exe'
+- name: IL2CPP_SUPPORT_URL
```
Comment (lucen-ms) [inline .azuredevops/appcenter-sdk-unity-build-release-1es.yml:11]: Where is this variable used?

File: .azuredevops/appcenter-sdk-unity-build-release-1es.yml [hunk 5]
```diff
@@ -4,6 +4,19 @@ trigger:
...
+  value: 'https://download.unity3d.com/download_unity/61fce66342ad/TargetSupportInstaller/UnitySetup-UWP-IL2CPP-Support-for-Editor-2018.4.18f1.exe'
+- name: NUGET_FEED_ID
+  value: 'AppCenter'
+- name: NUGET_USER
+  value: appcenter
+- name: AZURE_STORAGE_ACCOUNT
```
Comment (lucen-ms) [inline .azuredevops/appcenter-sdk-unity-build-release-1es.yml]: Where is this variable used?

Comment (AnatolyPristensky) [reply .azuredevops/appcenter-sdk-unity-build-release-1es.yml]: It used in build.cake script, that running in PublishPackagesToStorage task. 
https://github.com/microsoft/appcenter-sdk-unity/blob/b85e7669d71a38c304646038ad26bc3819ca10c5/build.cake#L677

File: .azuredevops/appcenter-sdk-unity-build-release-1es.yml (hunk 4 above)
Comment (AnatolyPristensky) [reply .azuredevops/appcenter-sdk-unity-build-release-1es.yml:11]: It used in build.cake script, that running in Install Unity task. 
https://github.com/microsoft/appcenter-sdk-unity/blob/b85e7669d71a38c304646038ad26bc3819ca10c5/build.cake#L378

File: .azuredevops/appcenter-sdk-unity-build-release-1es.yml (hunk 3 above)
Comment (AnatolyPristensky) [reply .azuredevops/appcenter-sdk-unity-build-release-1es.yml:9]: It used in build.cake script, that running in Install Unity task. 
https://github.com/microsoft/appcenter-sdk-unity/blob/b85e7669d71a38c304646038ad26bc3819ca10c5/build.cake#L377

File: .azuredevops/appcenter-sdk-unity-build-release-1es.yml (hunk 5 above)
Comment (lucen-ms) [reply .azuredevops/appcenter-sdk-unity-build-release-1es.yml]: I don't see this task in this pipeline. What do we push to the storage account?

Comment (AnatolyPristensky) [reply .azuredevops/appcenter-sdk-unity-build-release-1es.yml]: You are correct. Removed. Successful build: https://dev.azure.com/msmobilecenter/SDK/_build/results?buildId=1522401&view=logs&s=eb264492-b105-5667-1a96-783f82eb00ce

**Review by lucen-ms — APPROVED** [review]

File: .azuredevops/appcenter-sdk-unity-build-release-1es.yml (hunk 5 above)
Comment (DmitriyKirakosyan) [reply .azuredevops/appcenter-sdk-unity-build-release-1es.yml]: @AnatolyPristensky , please add task for publication to storage. Looks like you forgot it. This task would use build target `PublishPackagesToStorage`, which would use the discussed variable.

Comment (lucen-ms) [reply .azuredevops/appcenter-sdk-unity-build-release-1es.yml]: What do we push to the storage account?

Comment (DmitriyKirakosyan) [reply .azuredevops/appcenter-sdk-unity-build-release-1es.yml]: .unitypackage files. The same release artifacts which we upload to github release.

**Pull request merged**